### PR TITLE
split cache->insert from cache->put

### PR DESCRIPTION
**Pull request by icewind1991**
Makes it easier for a cache wrapper to only overwrite the new file behavior.

cc @PVince81 @nickvergessen

### CODE REVIEW

**Comment by icewind1991** [comment]
https://github.com/owncloud/core/pull/22071/files?w=1 makes review a lot easier than looking at however git decided to interpret the indention change

**Comment by nickvergessen** [comment]
Looks good to me :+1:

**Comment by rullzer** [comment]
Makes sense. :+1:

File: lib/public/files/cache/icache.php [hunk 1]
```diff
@@ -87,6 +87,18 @@ public function getFolderContentsById($fileId);
...
+	 * insert meta data for a new file or folder
+	 *
+	 * @param string $file
+	 * @param array $data
+	 *
+	 * @return int file id
+	 * @throws \RuntimeException
+	 * @since 9.0.0
+	 */
+	public function insert($file, array $data);
```
Comment (PVince81) [inline lib/public/files/cache/icache.php]: can you update the comment on "put" to clarify that it automatically does a insert or update ?

Comment (icewind1991) [inline lib/public/files/cache/icache.php]: done

**Comment by PVince81** [comment]
:+1:

**Pull request merged**